components/Modal/CreateBlueprint.js

```diff
@@ -232,7 +255,7 @@ class CreateBlueprintModal extends React.Component {
             id="create-blueprint-modal-create-button"
             type="button"
             className="btn btn-primary"
-            disabled={this.state.name == "" || this.state.errorNameDuplicate}
+            disabled={this.state.name.length === 0 || this.nameContainsError()}
```

Comment:
errorNameEmpty in nameContainesError should make the first check redundant

Reply:
When the modal initially opens nameContainsError will return false because we do not check for an error before the user interacts with the modal. However, the user should still be prevented from creating a blueprint with an empty name. This duplicate check is an easy fix to disable the button and still not show errors as soon as the modal opens.

Reply:
I was wrong. I forgot about the onBlur call to handleErrorNameInvalid.